### PR TITLE
unset http_proxy variable for knife (bsc#958716)

### PR DESCRIPTION
**Pull request by rsalevsky**

### CODE REVIEW

**Comment by vuntz** [comment]
This is unsetting it for the whole install process afterwards. Is this wanted? If yes, why not do it at the beginning? If no, then I guess you need to save the old value and set it again :-)

**Comment by rsalevsky** [comment]
I need to check if this is really this final solution, I just took this fix from the bugreport. Maybe @dirkmueller can give more information?

File: scripts/install-chef-suse.sh [hunk 1]
```diff
@@ -665,7 +665,16 @@ echo_summary "Performing initial chef-client run"
 service chef-client status &> /dev/null && service chef-client stop
 
 if ! [ -e ~/.chef/knife.rb -a -e ~/.chef/root.pem ]; then
-    yes '' | knife configure -i
+    # no_proxy is currently not suppoted in ruby see bsc#958716
+    # unset it for this call
+    if [[ -n $http_proxy ]]; then
+        local proxy = $http_proxy
```
Comment (jdsn) [inline scripts/install-chef-suse.sh]: syntax error, this must read:
```
local proxy=$http_proxy
```

**Comment by jdsn** [comment]
My proposal without code duplication, and a little more readable (IMHO):
```
proxy_save=$http_proxy
unset $http_proxy
yes '' | knife configure -i
if [[ $proxy_save ]] ; then
  http_proxy=$proxy_save
fi
```

File: scripts/install-chef-suse.sh [hunk 2]
```diff
@@ -665,7 +665,14 @@ echo_summary "Performing initial chef-client run"
 service chef-client status &> /dev/null && service chef-client stop
 
 if ! [ -e ~/.chef/knife.rb -a -e ~/.chef/root.pem ]; then
+    # no_proxy is currently not suppoted in ruby see bsc#958716
+    # unset it for this call
+    proxy_save=$http_proxy
+    unset http_proxy
     yes '' | knife configure -i
+    if [[ $proxy_save ]] ; then
+      http_proxy=$proxy_save
```
Comment (bmwiedemann) [inline scripts/install-chef-suse.sh]: travis complains about this line: Indent not multiple of 4

**Comment by bmwiedemann** [comment]
one error: unset http_proxy needs to be written without $
and then we could probably avoid the backup+restore of the value by using a subshell with
```
( unset http_proxy ; yes '' | knife configure -i )
```
or even without a sub-shell
```
yes '' | http_proxy= knife configure -
```

**Comment by rsalevsky** [comment]
@bmwiedemann done

File: scripts/install-chef-suse.sh [hunk 3]
```diff
@@ -665,7 +665,9 @@ echo_summary "Performing initial chef-client run"
 service chef-client status &> /dev/null && service chef-client stop
 
 if ! [ -e ~/.chef/knife.rb -a -e ~/.chef/root.pem ]; then
-    yes '' | knife configure -i
+    # no_proxy is currently not suppoted in ruby see bsc#958716
+    # unset it for this call
+    (unset http_proxy; yes '' | knife configure -i)
```
Comment (jdsn) [inline scripts/install-chef-suse.sh]: install-chef-suse.sh runs with `set -e`
Moving the knife call to a subshell would prevent the -e to exit install-chef-suse.sh if knife exits with an error
So $? needs to be evaluated right after the subshell and exit if it is != 0

**Comment by jdsn** [comment]
:+1:

File: scripts/install-chef-suse.sh [hunk 4]
```diff
@@ -665,7 +665,11 @@ echo_summary "Performing initial chef-client run"
...
+    ret=$?
+    [[ $ret != 0 ]] && exit $ret
```
Comment (bmwiedemann) [inline scripts/install-chef-suse.sh]: could also be done with  || exit $?

Comment (jdsn) [inline scripts/install-chef-suse.sh]: agreed

**Comment by bmwiedemann** [comment]
+1

File: scripts/install-chef-suse.sh [hunk 5]
```diff
@@ -665,7 +665,9 @@ echo_summary "Performing initial chef-client run"
 service chef-client status &> /dev/null && service chef-client stop
 
 if ! [ -e ~/.chef/knife.rb -a -e ~/.chef/root.pem ]; then
-    yes '' | knife configure -i
+    # no_proxy is currently not suppoted in ruby see bsc#958716
```
Comment (vuntz) [inline scripts/install-chef-suse.sh]: typo: supported

**Comment by vuntz** [comment]
+1

**Comment by mjura** [comment]
+1

**Comment by vuntz** [comment]
I think we're missing the backport to 3.0 for that one, aren't we?

**Comment by rsalevsky** [comment]
@vuntz see #2214

**Comment by vuntz** [comment]
> @vuntz see #2214

That's a backport to tex, not 3.0.

**Comment by rsalevsky** [comment]
@vuntz Ups, created a new PR see #2216

**Pull request merged**